src/TLuaInterpreter.cpp

```diff
@@ -18441,3 +18451,33 @@ void TLuaInterpreter::updateExtendedAnsiColorsInTable()
         lua_pop(L, 1);
     }
 }
+
+// Internal function - Looks for headers in an http response. If it finds them,
+// it creates a table on the lua stack for the headers, pops the table from the
+// stack, puts it in the lua registry, and returns the key to where it is in
+// registery wrapped in an optional. Otherwise it returns an empty optional.
+std::optional<int> TLuaInterpreter::createHttpHeadersTable(QNetworkReply* reply)
+{
+    lua_State* L = pGlobalLua;
+    if (!L) {
+        qDebug() << "LUA CRITICAL ERROR: no suitable Lua execution unit found.";
+        return {};
+    }
+    QList<QByteArray> headerList = reply->rawHeaderList();
+    if (headerList.size() <= 0) {
+        return {};
+    }
+
+    // Push table onto stack
+    lua_newtable(L);
+    for (QByteArray head : headerList) {
```

Comment:
I believe we need a [qAsConst()](https://doc.qt.io/qt-5/qtglobal.html#qAsConst) here so a copy isn't made and perhaps it's better to call it a `header`.

```suggestion
    for (QByteArray head : qAsConst(headerList)) {
```

Reply:
Done, thanks for the documentation. In it, I read that perhaps I should just declare `headerList` as a `const` if I can do that instead. Do you agree?

Reply:
Yep